indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisorReportPayload.java

```diff
@@ -0,0 +1,161 @@
+ */
+
+package org.apache.druid.indexing.seekablestream.supervisor;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.druid.java.util.common.IAE;
+import org.joda.time.DateTime;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public abstract class SeekableStreamSupervisorReportPayload<partitionType, sequenceType>
+{
+  private final String dataSource;
+  private final String id;
+  private final int partitions;
+  private final int replicas;
+  private final long durationSeconds;
+  private final List<TaskReportData> activeTasks;
+  private final List<TaskReportData> publishingTasks;
+  private final Map<partitionType, sequenceType> latestOffsets;
+  private final Map<partitionType, sequenceType> minimumLag;
+  private final Long aggregateLag;
+  private final DateTime offsetsLastUpdated;
+  private final boolean suspended;
+
+  public SeekableStreamSupervisorReportPayload(
+      String dataSource,
+      String id,
+      int partitions,
+      int replicas,
+      long durationSeconds,
+      @Nullable Map<partitionType, sequenceType> latestOffsets,
+      @Nullable Map<partitionType, sequenceType> minimumLag,
+      @Nullable Long aggregateLag,
+      @Nullable DateTime offsetsLastUpdated,
+      boolean suspended
+  )
+  {
+    this.dataSource = dataSource;
+    this.id = id;
+    this.partitions = partitions;
+    this.replicas = replicas;
+    this.durationSeconds = durationSeconds;
+    this.activeTasks = new ArrayList<>();
+    this.publishingTasks = new ArrayList<>();
+    this.latestOffsets = latestOffsets;
+    this.minimumLag = minimumLag;
+    this.aggregateLag = aggregateLag;
+    this.offsetsLastUpdated = offsetsLastUpdated;
+    this.suspended = suspended;
+  }
+
+  public void addTask(TaskReportData data)
+  {
+    if (data.getType().equals(TaskReportData.TaskType.ACTIVE)) {
+      activeTasks.add(data);
+    } else if (data.getType().equals(TaskReportData.TaskType.PUBLISHING)) {
+      publishingTasks.add(data);
+    } else {
+      throw new IAE("Unknown task type [%s]", data.getType().name());
+    }
+  }
+
+  @JsonProperty
+  public String getDataSource()
+  {
+    return dataSource;
+  }
+
+  public String getId()
+  {
+    return id;
+  }
+
+  @JsonProperty
+  public int getPartitions()
+  {
+    return partitions;
+  }
+
+  @JsonProperty
+  public int getReplicas()
+  {
+    return replicas;
+  }
+
+  @JsonProperty
+  public boolean getSuspended()
+  {
+    return suspended;
+  }
+
+  @JsonProperty
+  public long getDurationSeconds()
+  {
+    return durationSeconds;
+  }
+
+  @JsonProperty
+  public List<? extends TaskReportData> getActiveTasks()
+  {
+    return activeTasks;
+  }
+
+  @JsonProperty
+  public List<? extends TaskReportData> getPublishingTasks()
+  {
+    return publishingTasks;
+  }
+
+  @JsonProperty
+  @JsonInclude(JsonInclude.Include.NON_NULL)
```

Comment:
Probably would be better to just throw this ` @JsonInclude(JsonInclude.Include.NON_NULL)` annotation on the entire class